File: 3959-check-good-integer/3959-check-good-integer.php
class Solution {

    /**
     * @param Integer $n
     * @return Boolean
     */
    function checkGoodInteger($n) {
        $ds = 0;
        $ss = 0;

        while ($n > 0) {
            $curr = $n % 10;
            $ds += $curr;
            $ss += $curr * $curr;
            $n = intdiv($n, 10);
        }

        return $ss - $ds >= 50;
    }
}

File: 3959-check-good-integer/3959-check-good-integer.rb
# @param {Integer} n
# @return {Boolean}
def check_good_integer(n)
  ds = 0
  ss = 0

  while n > 0
    curr = n % 10
    ds += curr
    ss += curr * curr
    n /= 10
  end

  ss - ds >= 50
end